### ISSUE TITLE
TRAINING RF-DETRSegNano with yolo format takes up all the ram of the machine and crashes it before training start

### ISSUE THREAD

**Issue by Ahmedd-Wahdan** [roboflow/rf-detr#733]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

i tried to train the rf-detrSegNano model on custom dataset using yolo segmentation format and when i try to train the model the ram of the machine gets filled and the machine crashes , although the whole dataset is <8GB and the machine had 16GB of RAM and still it took the whole 16GB and crashed the machine

### Environment
```
OS:Ubuntu 24.04.4 LTS
rf-detr version: 1.5.0
python: 3.10.12
torch: 2.8.0
GPU: Tesla T4 16 gb Vram
```

### Minimal Reproducible Example
```py
from rfdetr import RFDETRSegNano


model = RFDETRSegNano()

model.train(
    dataset_dir="docdet_dataset",
    dataset_file="yolo" ,
    epochs=5,
    batch_size=2,
    grad_accum_steps=8,
    lr=1e-4,
    output_dir="output",
    device="cuda" ,
    gradient_checkpointing = True ,
    use_ema =True ,
    tensorboard=True,
    print_freq=140 , 
    run_test=False ,   
)
```

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

**Comment by Borda** [comment]
@Ahmedd-Wahdan, thank you for this finding. Could you please create and share a Colab notebook that includes downloading the dataset you used so we can have a fully reproducible example?

**Comment by Ahmedd-Wahdan** [comment]
i dont have the rights to share it pubublicly because it's private data , but even with small size dataset (3.3GB train and test) takes about 12GB of am when training the segmentation model ,(i think it has to do with the loading of the masks and operation done on it , also num_workers) ,  if you want i will try to look into it and submit a pr , because i will be working on training RF-DETR models more for the upcoming months

**Comment by laclouis5** [comment]
@Ahmedd-Wahdan The evaluation code of the segmentation model resizes the mask queries of the model to the original image size. If some of your validation images are large, the GPU memory will get very large.

For instance, RFDETRSegNano outputs 100 mask queries and if one of your input image is 12M pixels (3000x4000), then evaluation code will try to resize the output tensor to a shape of 100x3000x4000, with corresponds to 4.5GB of GPU memory in FP32.

In this situation, the simplest solution is to resize your dataset prior to training. A maximum side length of 1024 is usually sufficient for most applications and reduces significantly the memory usage. Do not forget to scale the annotations (bbox coordinates, segmentation polygons and area) accordingly.
